qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_uvcc_factory.py

```diff
@@ -29,31 +30,66 @@
 from qiskit_nature.properties.second_quantization.vibrational import (
     VibrationalStructureDriverResult,
 )
+from qiskit_nature.deprecation import deprecate_property, deprecate_positional_arguments
 
 from .minimum_eigensolver_factory import MinimumEigensolverFactory
 from ...initial_points import InitialPoint, VSCFInitialPoint
 
+logger = logging.getLogger(__name__)
 
-class VQEUVCCFactory(MinimumEigensolverFactory):
-    """A factory to construct a VQE minimum eigensolver with UVCCSD ansatz wavefunction."""
 
+class VQEUVCCFactory(MinimumEigensolverFactory):
+    """Factory to construct a :class:`VQE` minimum eigensolver with :class:`UVCCSD` ansatz wavefunction.
+
+    Note: Changing the ansatz of the minimum_eigensolver of the factory won't affect
+        the ansatz of new solvers produced by the factory. On top of that the default
+        type for :class:`VQE` is not the same as the factory.
+
+    .. code-block:: python
+
+        # Wrong way
+        factory = VQEUVCCFactory()
+        print(type(factory.minimum_eigensolver.ansatz))  # RealAmplitudes (by default)
+        factory.minimum_eigensolver.ansatz = UVCC()
+        print(type(factory.minimum_eigensolver.ansatz))  # UVCC
+        vqe = factory.get_solver(problem, qubit_converter)
+        print(type(vqe.ansatz))  # UVCCSD (and not UVCC nor RealAmplitudes)
```

Comment:
This "Wrong way" is basically just proof, that as the note above says, the factory overwrites any ansatz that the VQE may have. Do we need/want to show that in code here? Maybe if its clear that this is what is being shown - to me it was not. The intended way - is a simple example showing how to use the factory right.